src/ModalHeader.js

```diff
@@ -4,24 +4,35 @@ import classnames from 'classnames';
 
 class ModalHeader extends React.Component {
 
+  renderCloseButton() {
+    if (this.props.closeButton === true) {
+      return (
+        <button
+          className='close'
+          aria-label={this.props['aria-label'] || 'Close'} //eslint-disable-line react/prop-types
+          onClick={this.props.onHide}
+          style={{ marginTop: -2 }}
+        >
+          <span aria-hidden="true">
+            &times;
+          </span>
+        </button>
+      );
+    }
+    else if (React.isValidElement(this.props.closeButton)) {
```

Comment:
yeah I agree I  the case of a custom element it should just pass onHide as onHide, instead of as onClick.